src/ir_Corona.cpp

```diff
@@ -0,0 +1,578 @@
+void IRCoronaAc::setSwingV(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionData0Pos],
+         kCoronaAcSwingVToogleOffset, on);
+}
+
+/// Get the Vertical Swing mode of the A/C.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getSwingV(void) {
+  return GETBIT64(remote_state[kCoronaAcSectionData0Pos],
+                  kCoronaAcSwingVToogleOffset);
+}
+
+/// Set the Timer time
+/// @param[in] section index of section, used for offset.
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (non in range value is disable).
+///   Valid is from 1 minute to 12 hours
+void IRCoronaAc::_setTimer(const uint8_t section, const uint16_t nr_of_mins) {
+  // default to off
+  uint16_t hsecs = kCoronaAcTimerOff;
+  if (1 <= nr_of_mins && nr_of_mins <= kCoronaAcTimerMax)
+    hsecs = nr_of_mins * kCoronaAcTimerUnitsPerMin;
+
+  uint8_t pos = section * kCoronaAcSectionBytes;
+  remote_state[pos + kCoronaAcSectionData1Pos] = (hsecs & 0xff00) >> 8;
+  remote_state[pos + kCoronaAcSectionData0Pos] = (hsecs & 0xff);
+}
+
+/// Get the current Timer time
+/// @return The number of minutes it is set for. 0 means it's off.
+/// @note The A/C protocol supports 2 second increments
+uint16_t IRCoronaAc::_getTimer(const uint8_t section) {
+  uint8_t pos = section * kCoronaAcSectionBytes;
+  uint16_t hsecs = remote_state[pos + kCoronaAcSectionData1Pos] << 8 |
+                   remote_state[pos + kCoronaAcSectionData0Pos];
+
+  if (hsecs == kCoronaAcTimerOff)
+    return 0;
+
+  return hsecs / kCoronaAcTimerUnitsPerMin;
+}
+
+/// Get the current On Timer time
+/// @return The number of minutes it is set for. 0 means it's off.
+uint16_t IRCoronaAc::getOnTimer(void) {
+  return _getTimer(kCoronaAcOnTimerSection);
+}
+
+/// Set the On Timer time
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (0 or kCoronaAcTimerOff is disable).
+void IRCoronaAc::setOnTimer(const uint16_t nr_of_mins) {
+  _setTimer(kCoronaAcOnTimerSection, nr_of_mins);
+  // if we set a timer value, clear the other timer
+  if (getOnTimer())
+    setOffTimer(kCoronaAcTimerOff);
+}
+
+/// Get the current Off Timer time
+/// @return The number of minutes it is set for. 0 means it's off.
+uint16_t IRCoronaAc::getOffTimer(void) {
+  return _getTimer(kCoronaAcOffTimerSection);
+}
+
+/// Set the Off Timer time
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (0 or kCoronaAcTimerOff is disable).
+void IRCoronaAc::setOffTimer(const uint16_t nr_of_mins) {
+  _setTimer(kCoronaAcOffTimerSection, nr_of_mins);
+  // if we set a timer value, clear the other timer
+  if (getOffTimer())
+    setOnTimer(kCoronaAcTimerOff);
+}
+
+/// Convert the internal state into a human readable string.
+/// @return The current internal state expressed as a human readable String.
+String IRCoronaAc::toString(void) {
+  String result = "";
+  result.reserve(120);  // Reserve some heap for the string to reduce fragging.
```

Comment:
Bump value to `140`, I grabbed one of your strings, and it was already 133 chars long.